Issues with pyright-langserver

### Summary

Using the pyright-langserver for my linter results in async timeout error whenever I do anything.

### Reproduction Steps

Install the pyright langsever as described on the wiki page.

This is my global config.
```toml
[[language]]
name = "python"
roots = ["pyproject.toml"]
language-server = { command = "/usr/local/bin/pyright-langserver", args = ["--stdio"] }
config = {} # <- this is the important line

formatter = { command = "black", args = ["--quiet", "-"]}
auto-format = true
```

### Helix log

Not sure where the exact error was so I put everything.

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-02-10T20:12:43.970 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["[scratch]"]
2023-02-10T20:13:24.302 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-02-10T20:13:48.899 helix_view::editor [ERROR] editor error: Could not load theme: No such file or directory (os error 2)
2023-02-10T20:14:48.424 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:14:55.819 helix_term::application [WARN] failed to load theme `monokai` - No such file or directory (os error 2)
2023-02-10T20:14:55.820 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:15:17.382 helix_term::application [WARN] failed to load theme `monokai` - No such file or directory (os error 2)
2023-02-10T20:15:19.984 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-02-10T20:19:31.972 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-02-10T20:19:48.649 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:20:44.265 helix_view::editor [ERROR] editor error: no last accessed buffer
2023-02-10T20:21:43.280 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:29:19.745 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-02-10T20:29:25.183 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-02-10T20:33:09.364 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-02-10T20:35:24.915 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:39:24.922 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:39:41.844 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:39:51.057 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:40:24.361 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:40:54.960 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:43:41.260 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-02-10T20:44:24.100 helix_lsp::transport [ERROR] err <- "Ignoring notification for unknown method \"workspace/didChangeConfiguration\"\n"
2023-02-10T20:45:17.068 helix_lsp::client [INFO] Using custom LSP config: {}
2023-02-10T20:45:17.068 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":false},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"publishDiagnostics":{},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":false},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"executeCommand":{"dynamicRegistration":false},"symbol":{"dynamicRegistration":false},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"22.12 (6929a12f)"},"initializationOptions":{},"processId":116113,"rootPath":"/home/lunar/Documents/Projects/Python/code-runner","rootUri":"file:///home/lunar/Documents/Projects/Python/code-runner","workspaceFolders":[{"name":"code-runner","uri":"file:///home/lunar/Documents/Projects/Python/code-runner"}]},"id":0}
2023-02-10T20:45:17.419 helix_lsp::transport [INFO] <- {"id": 0, "jsonrpc": "2.0", "result": {"capabilities": {"textDocumentSync": {"openClose": true, "change": 2, "save": true}, "hoverProvider": true, "codeActionProvider": {"codeActionKinds": ["quickfix", "source.fixAll", "source.organizeImports", "CodeActionKind.SourceFixAll.ruff", "CodeActionKind.SourceOrganizeImports.ruff"], "resolveProvider": true}, "executeCommandProvider": {"commands": ["ruff.applyAutofix", "ruff.applyOrganizeImports"]}, "workspace": {"workspaceFolders": {"supported": true, "changeNotifications": true}, "fileOperations": {}}}, "serverInfo": {"name": "Ruff", "version": "0.0.17"}}}
2023-02-10T20:45:17.419 helix_lsp::transport [INFO] <- {"capabilities":{"codeActionProvider":{"codeActionKinds":["quickfix","source.fixAll","source.organizeImports","CodeActionKind.SourceFixAll.ruff","CodeActionKind.SourceOrganizeImports.ruff"],"resolveProvider":true},"executeCommandProvider":{"commands":["ruff.applyAutofix","ruff.applyOrganizeImports"]},"hoverProvider":true,"textDocumentSync":{"change":2,"openClose":true,"save":true},"workspace":{"fileOperations":{},"workspaceFolders":{"changeNotifications":true,"supported":true}}},"serverInfo":{"name":"Ruff","version":"0.0.17"}}
2023-02-10T20:45:17.419 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialized","params":{}}
2023-02-10T20:45:17.419 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"workspace/didChangeConfiguration","params":{"settings":{}}}
2023-02-10T20:45:17.419 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"languageId":"python","text":"TOKEN=\"REDACTED LOL\"\nPISTON=\"https://emkc.org/api/v2/piston/\"\n\n\n\"MTA3Mzc3MTY1ODkwNjcwMTk1NA.GbYAok.6WkybZrDzwgHL1JWLJeR2FMvtrX2gslhNNm5dk\"\n","uri":"file:///home/lunar/Documents/Projects/Python/code-runner/config.py","version":0}}}
2023-02-10T20:45:17.422 helix_lsp::transport [ERROR] err <- "Ignoring notification for unknown method \"workspace/didChangeConfiguration\"\n"
2023-02-10T20:45:17.424 helix_lsp::transport [INFO] <- {"params": {"type": 4, "message": "Using interpreter executable: /home/lunar/.pyenv/versions/3.11.1/bin/ruff"}, "method": "window/logMessage", "jsonrpc": "2.0"}
2023-02-10T20:45:17.424 helix_lsp::transport [INFO] <- {"params": {"type": 4, "message": "Running Ruff with: ['/home/lunar/.pyenv/versions/3.11.1/bin/ruff', '--no-cache', '--no-fix', '--quiet', '--format', 'json', '-', '--stdin-filename', '/home/lunar/Documents/Projects/Python/code-runner/config.py']"}, "method": "window/logMessage", "jsonrpc": "2.0"}
2023-02-10T20:45:17.425 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "Using interpreter executable: /home/lunar/.pyenv/versions/3.11.1/bin/ruff" }
2023-02-10T20:45:17.425 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "Running Ruff with: ['/home/lunar/.pyenv/versions/3.11.1/bin/ruff', '--no-cache', '--no-fix', '--quiet', '--format', 'json', '-', '--stdin-filename', '/home/lunar/Documents/Projects/Python/code-runner/config.py']" }
2023-02-10T20:45:17.431 helix_lsp::transport [INFO] <- {"params": {"uri": "file:///home/lunar/Documents/Projects/Python/code-runner/config.py", "diagnostics": []}, "method": "textDocument/publishDiagnostics", "jsonrpc": "2.0"}
2023-02-10T20:45:18.644 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"shutdown","id":1}
2023-02-10T20:45:18.647 helix_lsp::transport [INFO] <- {"id": 1, "jsonrpc": "2.0", "result": null}
2023-02-10T20:45:18.648 helix_lsp::transport [INFO] <- null
2023-02-10T20:45:18.648 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"exit"}
2023-02-10T20:46:28.804 helix_lsp::transport [ERROR] err <- "Ignoring notification for unknown method \"workspace/didChangeConfiguration\"\n"
2023-02-10T20:50:18.503 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:50:23.068 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T20:52:14.228 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-10T20:53:56.293 helix_view::editor [ERROR] editor error: Could not load theme: No such file or directory (os error 2)
2023-02-10T20:56:58.749 helix_view::document [ERROR] Formatter error: error: cannot format -: Cannot parse: 21:9:         )i

2023-02-10T21:00:49.711 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-10T21:01:12.589 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-10T21:03:40.184 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T21:03:56.407 helix_view::editor [ERROR] editor error: can't save file, parent directory does not exist
2023-02-10T21:03:58.385 helix_view::editor [ERROR] editor error: can't save file, parent directory does not exist
2023-02-10T21:03:58.385 helix_view::editor [ERROR] editor error: can't save file, parent directory does not exist
2023-02-10T21:04:01.347 helix_view::editor [ERROR] editor error: no such command: 'fW'
2023-02-10T21:04:03.076 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: [".helix/languages.toml"]
2023-02-10T21:04:05.426 helix_view::editor [ERROR] editor error: can't save file, parent directory does not exist
2023-02-10T21:04:14.931 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T21:04:28.125 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-10T21:05:19.411 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T21:06:09.231 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-10T21:07:09.749 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T21:07:49.129 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-10T21:08:28.809 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T21:08:31.005 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T21:08:35.142 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T21:09:30.506 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:09:30.508 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:09:51.975 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["bot/model.py"]
2023-02-10T21:09:53.433 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:09:54.235 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:09:54.587 helix_view::editor [ERROR] editor error: no such command: 'qw!'
2023-02-10T21:09:56.451 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(1)), original request likely timed out
2023-02-10T21:09:56.452 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(2)), original request likely timed out
2023-02-10T21:11:45.704 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:11:45.706 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:12:11.559 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:12:11.812 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:12:12.721 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:12:20.254 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(1)), original request likely timed out
2023-02-10T21:12:20.255 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(2)), original request likely timed out
2023-02-10T21:12:20.255 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(3)), original request likely timed out
2023-02-10T21:12:24.604 helix_view::editor [ERROR] editor error: Register [@] empty
2023-02-10T21:13:08.939 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:13:08.940 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:13:11.072 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:13:11.080 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-02-10T21:13:12.351 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:13:12.352 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:13:16.820 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:13:16.827 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-02-10T21:13:18.105 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:13:18.107 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:14:18.572 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-10T21:14:56.831 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:14:56.832 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:15:06.405 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:15:06.416 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-02-10T21:15:07.567 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:15:07.568 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:16:11.356 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:16:14.026 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:16:46.080 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:16:47.065 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:16:47.693 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:16:48.338 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:16:54.096 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:16:55.360 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:17:05.568 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:17:30.246 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:17:33.462 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:17:34.730 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:17:53.735 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:17:54.983 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:17:56.555 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:17:59.156 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(1)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(2)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(3)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(4)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(5)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(6)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(7)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(8)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(9)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(10)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(11)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(12)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(13)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(14)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(15)), original request likely timed out
2023-02-10T21:18:29.364 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(16)), original request likely timed out
2023-02-10T21:18:32.297 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:18:32.300 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:19:37.505 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:19:37.505 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:19:55.264 helix_view::theme [WARN] Theme: malformed hexcode: light-black
2023-02-10T21:19:55.264 helix_view::theme [WARN] Theme: malformed hexcode: light-black
2023-02-10T21:19:55.264 helix_view::theme [WARN] Theme: malformed hexcode: purple
2023-02-10T21:22:11.022 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-10T21:22:11.023 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-10T21:22:32.851 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:35.721 helix_view::document [ERROR] Formatter error: error: cannot format -: Cannot parse: 17:4:     return cls(

2023-02-10T21:22:35.778 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:37.131 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:37.423 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:37.547 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:37.669 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:37.786 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:37.935 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:38.069 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:38.269 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:38.278 helix_view::document [ERROR] Formatter error: error: cannot format -: Cannot parse: 17:4:     return cls(

2023-02-10T21:22:40.037 helix_view::document [ERROR] Formatter error: error: cannot format -: Cannot parse: 17:4:     return cls(

2023-02-10T21:22:40.083 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:53.948 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:22:54.165 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:23:04.396 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:23:06.271 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:23:06.876 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:23:09.429 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:23:10.573 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:23:12.040 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:23:27.304 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-10T21:26:03.064 helix_view::editor [ERROR] editor error: Register [@] empty
2023-02-10T21:26:05.020 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(1)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(2)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(3)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(4)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(5)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(6)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(7)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(8)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(9)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(10)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(11)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(12)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(13)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(15)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(16)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(17)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(18)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(19)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(14)), original request likely timed out
2023-02-10T21:26:05.021 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(20)), original request likely timed out
2023-02-12T12:36:28.103 helix_view::editor [ERROR] Failed to initialize the LSP for `source.bash` { cannot find binary path }
2023-02-12T16:48:10.044 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.basic` { cannot find binary path }
2023-02-12T16:48:19.739 helix_view::editor [ERROR] Failed to initialize the LSP for `source.java` { cannot find binary path }
2023-02-12T16:55:29.824 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-12T16:55:35.695 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-12T16:55:48.564 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-12T16:55:55.642 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-12T16:57:27.243 helix_view::editor [ERROR] Failed to initialize the LSP for `source.js` { cannot find binary path }
2023-02-12T16:57:34.371 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { IO Error: No such file or directory (os error 2) }
2023-02-12T16:58:14.722 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-12T16:58:14.722 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-12T16:58:42.849 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T16:58:45.643 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T16:58:47.853 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:01:12.613 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:01:16.183 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:01:17.050 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:01:17.141 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:01:20.146 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:03:41.981 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:10:19.028 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-12T17:10:56.039 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-12T17:10:56.040 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-12T17:10:58.945 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-12T17:10:58.946 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-12T17:11:20.128 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:11:31.704 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-02-12T17:11:46.167 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:12:55.116 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(2)), original request likely timed out
2023-02-12T17:12:55.116 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(1)), original request likely timed out
2023-02-12T17:12:55.913 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-12T17:12:55.914 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-12T17:13:27.746 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:13:28.147 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:13:29.246 helix_view::editor [ERROR] editor error: Async job failed: request timed out
2023-02-12T17:14:27.081 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
```

</details>


### Platform

Linux

### Terminal Emulator

Alacritty

### Helix Version

helix 22.12 (6929a12f)

I see this behavior on pyright 1.1.293 and 1.1.294 (latest) but not 1.1.292.

<details><summary>1.1.292 works well... (log)</summary>

```
2023-02-19T14:23:20.982 helix_lsp::client [INFO] Using custom LSP config: {}
2023-02-19T14:23:20.983 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-32","utf-8","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":false,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"publishDiagnostics":{},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":false},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"executeCommand":{"dynamicRegistration":false},"symbol":{"dynamicRegistration":false},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"22.12 (31b0c758)"},"initializationOptions":{},"processId":85583,"rootPath":"/home/michael/tmp","rootUri":"file:///home/michael/tmp","workspaceFolders":[{"name":"tmp","uri":"file:///home/michael/tmp"}]},"id":0}
2023-02-19T14:23:21.100 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Pyright language server 1.1.292 starting"}}
2023-02-19T14:23:21.101 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Server root directory: /home/michael/tmp/node_modules/pyright/dist/"}}
2023-02-19T14:23:21.101 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Pyright language server 1.1.292 starting" }
2023-02-19T14:23:21.104 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Starting service instance \"tmp\""}}
2023-02-19T14:23:21.104 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":0,"result":{"capabilities":{"textDocumentSync":2,"definitionProvider":{"workDoneProgress":true},"declarationProvider":{"workDoneProgress":true},"typeDefinitionProvider":{"workDoneProgress":true},"referencesProvider":{"workDoneProgress":true},"documentSymbolProvider":{"workDoneProgress":true},"workspaceSymbolProvider":{"workDoneProgress":true},"hoverProvider":{"workDoneProgress":true},"documentHighlightProvider":{"workDoneProgress":true},"renameProvider":{"prepareProvider":true,"workDoneProgress":true},"completionProvider":{"triggerCharacters":[".","["],"resolveProvider":true,"workDoneProgress":true,"completionItem":{"labelDetailsSupport":true}},"signatureHelpProvider":{"triggerCharacters":["(",",",")"],"workDoneProgress":true},"codeActionProvider":{"codeActionKinds":["quickfix","source.organizeImports"],"workDoneProgress":true},"executeCommandProvider":{"commands":[],"workDoneProgress":true},"callHierarchyProvider":true}}}
2023-02-19T14:23:21.104 helix_lsp::transport [INFO] <- {"capabilities":{"callHierarchyProvider":true,"codeActionProvider":{"codeActionKinds":["quickfix","source.organizeImports"],"workDoneProgress":true},"completionProvider":{"completionItem":{"labelDetailsSupport":true},"resolveProvider":true,"triggerCharacters":[".","["],"workDoneProgress":true},"declarationProvider":{"workDoneProgress":true},"definitionProvider":{"workDoneProgress":true},"documentHighlightProvider":{"workDoneProgress":true},"documentSymbolProvider":{"workDoneProgress":true},"executeCommandProvider":{"commands":[],"workDoneProgress":true},"hoverProvider":{"workDoneProgress":true},"referencesProvider":{"workDoneProgress":true},"renameProvider":{"prepareProvider":true,"workDoneProgress":true},"signatureHelpProvider":{"triggerCharacters":["(",",",")"],"workDoneProgress":true},"textDocumentSync":2,"typeDefinitionProvider":{"workDoneProgress":true},"workspaceSymbolProvider":{"workDoneProgress":true}}}
2023-02-19T14:23:21.105 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialized","params":{}}
2023-02-19T14:23:21.106 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":0,"method":"client/registerCapability","params":{"registrations":[{"id":"c5af1f33-bd3e-448a-b32c-b163782c2a06","method":"workspace/didChangeWorkspaceFolders","registerOptions":{}}]}}
2023-02-19T14:23:21.109 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Server root directory: /home/michael/tmp/node_modules/pyright/dist/" }
2023-02-19T14:23:21.109 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Starting service instance \"tmp\"" }
2023-02-19T14:23:21.109 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-19T14:23:21.109 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"workspace/didChangeConfiguration","params":{"settings":{}}}
2023-02-19T14:23:21.109 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"languageId":"python","text":"import json\n\ndef func(a):\n    print(a)\n","uri":"file:///home/michael/tmp/f.py","version":0}}}
2023-02-19T14:23:21.110 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":1,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/michael/tmp","section":"python"}]}}
2023-02-19T14:23:21.110 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","result":[null],"id":1}
2023-02-19T14:23:21.111 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":2,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/michael/tmp","section":"python.analysis"}]}}
2023-02-19T14:23:21.111 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","result":[null],"id":2}
2023-02-19T14:23:21.111 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":3,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/michael/tmp","section":"pyright"}]}}
2023-02-19T14:23:21.111 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","result":[null],"id":3}
2023-02-19T14:23:21.135 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"No configuration file found."}}
2023-02-19T14:23:21.135 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "No configuration file found." }
2023-02-19T14:23:21.135 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"No pyproject.toml file found."}}
2023-02-19T14:23:21.135 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":2,"message":"stubPath /home/michael/tmp/typings is not a valid directory."}}
2023-02-19T14:23:21.135 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Assuming Python platform Linux"}}
2023-02-19T14:23:21.135 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Searching for source files"}}
2023-02-19T14:23:21.136 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Found 1 source file"}}
2023-02-19T14:23:21.143 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "No pyproject.toml file found." }
2023-02-19T14:23:21.143 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Warning, message: "stubPath /home/michael/tmp/typings is not a valid directory." }
2023-02-19T14:23:21.143 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Assuming Python platform Linux" }
2023-02-19T14:23:21.143 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Searching for source files" }
2023-02-19T14:23:21.143 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Found 1 source file" }
2023-02-19T14:23:21.548 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"uri":"file:///home/michael/tmp/f.py","version":0,"diagnostics":[]}}
2023-02-19T14:23:24.534 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/signatureHelp","params":{"position":{"character":10,"line":3},"textDocument":{"uri":"file:///home/michael/tmp/f.py"}},"id":1}
2023-02-19T14:23:24.681 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":1,"result":{"signatures":[{"label":"(*values: object, sep: str | None = ..., end: str | None = ..., file: SupportsWrite[str] | None = ..., flush: Literal[False] = ...) -> None","parameters":[{"label":[1,16]},{"label":[18,39]},{"label":[41,62]},{"label":[64,101]},{"label":[103,130]}],"activeParameter":0},{"label":"(*values: object, sep: str | None = ..., end: str | None = ..., file: _SupportsWriteAndFlush[str] | None = ..., flush: bool) -> None","parameters":[{"label":[1,16]},{"label":[18,39]},{"label":[41,62]},{"label":[64,110]},{"label":[112,123]}],"activeParameter":0}],"activeSignature":0,"activeParameter":5}}
2023-02-19T14:23:24.681 helix_lsp::transport [INFO] <- {"activeParameter":5,"activeSignature":0,"signatures":[{"activeParameter":0,"label":"(*values: object, sep: str | None = ..., end: str | None = ..., file: SupportsWrite[str] | None = ..., flush: Literal[False] = ...) -> None","parameters":[{"label":[1,16]},{"label":[18,39]},{"label":[41,62]},{"label":[64,101]},{"label":[103,130]}]},{"activeParameter":0,"label":"(*values: object, sep: str | None = ..., end: str | None = ..., file: _SupportsWriteAndFlush[str] | None = ..., flush: bool) -> None","parameters":[{"label":[1,16]},{"label":[18,39]},{"label":[41,62]},{"label":[64,110]},{"label":[112,123]}]}]}
2023-02-19T14:23:26.255 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"shutdown","id":2}
2023-02-19T14:23:26.256 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":2,"result":null}
2023-02-19T14:23:26.256 helix_lsp::transport [INFO] <- null
2023-02-19T14:23:26.256 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"exit"}
```

</details>

<details><summary>But 1.1.294 seems to get stuck... (log)</summary>

```
2023-02-19T15:28:04.667 helix_lsp::client [INFO] Using custom LSP config: {}
2023-02-19T15:28:04.668 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-32","utf-8","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":false,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"publishDiagnostics":{},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":false},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"executeCommand":{"dynamicRegistration":false},"symbol":{"dynamicRegistration":false},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"22.12 (31b0c758)"},"initializationOptions":{},"processId":100190,"rootPath":"/home/michael/tmp","rootUri":"file:///home/michael/tmp","workspaceFolders":[{"name":"tmp","uri":"file:///home/michael/tmp"}]},"id":0}
2023-02-19T15:28:04.786 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Pyright language server 1.1.294 starting"}}
2023-02-19T15:28:04.787 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Server root directory: /home/michael/tmp/node_modules/pyright/dist/"}}
2023-02-19T15:28:04.787 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Pyright language server 1.1.294 starting" }
2023-02-19T15:28:04.790 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Starting service instance \"tmp\""}}
2023-02-19T15:28:04.791 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":0,"result":{"capabilities":{"textDocumentSync":2,"definitionProvider":{"workDoneProgress":true},"declarationProvider":{"workDoneProgress":true},"typeDefinitionProvider":{"workDoneProgress":true},"referencesProvider":{"workDoneProgress":true},"documentSymbolProvider":{"workDoneProgress":true},"workspaceSymbolProvider":{"workDoneProgress":true},"hoverProvider":{"workDoneProgress":true},"documentHighlightProvider":{"workDoneProgress":true},"renameProvider":{"prepareProvider":true,"workDoneProgress":true},"completionProvider":{"triggerCharacters":[".","["],"resolveProvider":true,"workDoneProgress":true,"completionItem":{"labelDetailsSupport":true}},"signatureHelpProvider":{"triggerCharacters":["(",",",")"],"workDoneProgress":true},"codeActionProvider":{"codeActionKinds":["quickfix","source.organizeImports"],"workDoneProgress":true},"executeCommandProvider":{"commands":[],"workDoneProgress":true},"callHierarchyProvider":true}}}
2023-02-19T15:28:04.791 helix_lsp::transport [INFO] <- {"capabilities":{"callHierarchyProvider":true,"codeActionProvider":{"codeActionKinds":["quickfix","source.organizeImports"],"workDoneProgress":true},"completionProvider":{"completionItem":{"labelDetailsSupport":true},"resolveProvider":true,"triggerCharacters":[".","["],"workDoneProgress":true},"declarationProvider":{"workDoneProgress":true},"definitionProvider":{"workDoneProgress":true},"documentHighlightProvider":{"workDoneProgress":true},"documentSymbolProvider":{"workDoneProgress":true},"executeCommandProvider":{"commands":[],"workDoneProgress":true},"hoverProvider":{"workDoneProgress":true},"referencesProvider":{"workDoneProgress":true},"renameProvider":{"prepareProvider":true,"workDoneProgress":true},"signatureHelpProvider":{"triggerCharacters":["(",",",")"],"workDoneProgress":true},"textDocumentSync":2,"typeDefinitionProvider":{"workDoneProgress":true},"workspaceSymbolProvider":{"workDoneProgress":true}}}
2023-02-19T15:28:04.791 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialized","params":{}}
2023-02-19T15:28:04.792 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":0,"method":"client/registerCapability","params":{"registrations":[{"id":"ceb366b5-5e34-46b2-8165-29403216fbb2","method":"workspace/didChangeWorkspaceFolders","registerOptions":{}}]}}
2023-02-19T15:28:04.795 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Server root directory: /home/michael/tmp/node_modules/pyright/dist/" }
2023-02-19T15:28:04.795 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Starting service instance \"tmp\"" }
2023-02-19T15:28:04.795 helix_term::application [ERROR] Language Server: Method not found client/registerCapability
2023-02-19T15:28:04.795 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"workspace/didChangeConfiguration","params":{"settings":{}}}
2023-02-19T15:28:04.795 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"languageId":"python","text":"import json\n\ndef func(a):\n    print(a)\n","uri":"file:///home/michael/tmp/f.py","version":0}}}
2023-02-19T15:28:04.796 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":1,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/michael/tmp","section":"python"}]}}
2023-02-19T15:28:04.796 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Starting service instance \"<default>\""}}
2023-02-19T15:28:04.796 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","result":[null],"id":1}
2023-02-19T15:28:04.796 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":2,"method":"workspace/configuration","params":{"items":[{"scopeUri":"","section":"python"}]}}
2023-02-19T15:28:04.796 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Starting service instance \"<default>\"" }
2023-02-19T15:28:04.797 helix_term::application [ERROR] received malformed method call from Language Server: workspace/configuration: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
2023-02-19T15:28:04.797 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":3,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/michael/tmp","section":"python.analysis"}]}}
2023-02-19T15:28:04.797 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","result":[null],"id":3}
2023-02-19T15:28:04.797 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":4,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/michael/tmp","section":"pyright"}]}}
2023-02-19T15:28:04.798 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","result":[null],"id":4}
2023-02-19T15:28:04.820 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"No configuration file found."}}
2023-02-19T15:28:04.820 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"No pyproject.toml file found."}}
2023-02-19T15:28:04.820 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "No configuration file found." }
2023-02-19T15:28:04.820 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":2,"message":"stubPath /home/michael/tmp/typings is not a valid directory."}}
2023-02-19T15:28:04.820 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Assuming Python platform Linux"}}
2023-02-19T15:28:04.821 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Searching for source files"}}
2023-02-19T15:28:04.821 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Found 1 source file"}}
2023-02-19T15:28:04.828 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "No pyproject.toml file found." }
2023-02-19T15:28:04.828 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Warning, message: "stubPath /home/michael/tmp/typings is not a valid directory." }
2023-02-19T15:28:04.828 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Assuming Python platform Linux" }
2023-02-19T15:28:04.828 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Searching for source files" }
2023-02-19T15:28:04.828 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Found 1 source file" }
2023-02-19T15:28:08.629 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/signatureHelp","params":{"position":{"character":10,"line":3},"textDocument":{"uri":"file:///home/michael/tmp/f.py"}},"id":1}
2023-02-19T15:28:12.317 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"shutdown","id":2}
2023-02-19T15:28:12.673 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":2,"result":null}
2023-02-19T15:28:12.674 helix_lsp::transport [INFO] <- null
2023-02-19T15:28:12.674 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":1,"result":{"signatures":[{"label":"(*values: object, sep: str | None = \" \", end: str | None = \"\\n\", file: SupportsWrite[str] | None = None, flush: Literal[False] = False) -> None","parameters":[{"label":[1,16]},{"label":[18,39]},{"label":[41,63]},{"label":[65,103]},{"label":[105,134]}],"activeParameter":0},{"label":"(*values: object, sep: str | None = \" \", end: str | None = \"\\n\", file: _SupportsWriteAndFlush[str] | None = None, flush: bool) -> None","parameters":[{"label":[1,16]},{"label":[18,39]},{"label":[41,63]},{"label":[65,112]},{"label":[114,125]}],"activeParameter":0}],"activeSignature":0,"activeParameter":5}}
2023-02-19T15:28:12.674 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"exit"}
2023-02-19T15:28:12.674 helix_lsp::transport [INFO] <- {"activeParameter":5,"activeSignature":0,"signatures":[{"activeParameter":0,"label":"(*values: object, sep: str | None = \" \", end: str | None = \"\\n\", file: SupportsWrite[str] | None = None, flush: Literal[False] = False) -> None","parameters":[{"label":[1,16]},{"label":[18,39]},{"label":[41,63]},{"label":[65,103]},{"label":[105,134]}]},{"activeParameter":0,"label":"(*values: object, sep: str | None = \" \", end: str | None = \"\\n\", file: _SupportsWriteAndFlush[str] | None = None, flush: bool) -> None","parameters":[{"label":[1,16]},{"label":[18,39]},{"label":[41,63]},{"label":[65,112]},{"label":[114,125]}]}]}
```

</details>

It looks like 1.1.293 started sending a `workspace/configuration` request with an invalid scopeUri: an empty string (which is an invalid URI type). We currently log that error but don't reply to the rpc request: https://github.com/helix-editor/helix/blob/31b0c75832c56179e9fcb2301f1e92ac2fa116ff/helix-term/src/application.rs#L960-L967

Replying to that request with an error response fixes the hang for me. I'll open a PR.

--- a/helix-term/src/application.rs
+++ b/helix-term/src/application.rs
@@ -951,24 +951,32 @@ impl Application {
             Call::MethodCall(helix_lsp::jsonrpc::MethodCall {
                 method, params, id, ..
             }) => {
-                let call = match MethodCall::parse(&method, params) {
-                    Ok(call) => call,
+                let reply = match MethodCall::parse(&method, params) {
                     Err(helix_lsp::Error::Unhandled) => {
-                        error!("Language Server: Method not found {}", method);
-                        return;
+                        error!(
+                            "Language Server: Method {} not found in request {}",
+                            method, id
+                        );
+                        Err(helix_lsp::jsonrpc::Error {
+                            code: helix_lsp::jsonrpc::ErrorCode::MethodNotFound,
+                            message: format!("Method not found: {}", method),
+                            data: None,
+                        })
                     }
                     Err(err) => {
                         log::error!(
-                            "received malformed method call from Language Server: {}: {}",
+                            "Language Server: Received malformed method call {} in request {}: {}",
                             method,
+                            id,
                             err
                         );
-                        return;
+                        Err(helix_lsp::jsonrpc::Error {
+                            code: helix_lsp::jsonrpc::ErrorCode::ParseError,
+                            message: format!("Malformed method call: {}", method),
+                            data: None,
+                        })
                     }
-                };
-
-                let reply = match call {
-                    MethodCall::WorkDoneProgressCreate(params) => {
+                    Ok(MethodCall::WorkDoneProgressCreate(params)) => {
                         self.lsp_progress.create(server_id, params.token);
 
                         let editor_view = self
@@ -982,7 +990,7 @@ impl Application {
 
                         Ok(serde_json::Value::Null)
                     }
-                    MethodCall::ApplyWorkspaceEdit(params) => {
+                    Ok(MethodCall::ApplyWorkspaceEdit(params)) => {
                         apply_workspace_edit(
                             &mut self.editor,
                             helix_lsp::OffsetEncoding::Utf8,
@@ -995,13 +1003,13 @@ impl Application {
                             failed_change: None,
                         }))
                     }
-                    MethodCall::WorkspaceFolders => {
+                    Ok(MethodCall::WorkspaceFolders) => {
                         let language_server =
                             self.editor.language_servers.get_by_id(server_id).unwrap();
 
                         Ok(json!(language_server.workspace_folders()))
                     }
-                    MethodCall::WorkspaceConfiguration(params) => {
+                    Ok(MethodCall::WorkspaceConfiguration(params)) => {
                         let result: Vec<_> = params
                             .items
                             .iter()

